test/TransitionGroup-test.js

```diff
@@ -92,14 +92,24 @@ describe('TransitionGroup', () => {
     act(() => {
       jest.runAllTimers();
     });
-    expect(log).toEqual(['appear', 'appearing', 'appeared']);
+    expect(log).toEqual(
+      // React 18 StrictEffects will call `componentDidMount` twice causing two `onEnter` calls.
```

Comment:
See https://github.com/reactwg/react-18/discussions/19

Reply:
👍  I hope React has a way to detect the StrictEffects mode. 😄